pkg/network/tracer/tracer_test.go

```diff
@@ -1018,11 +1018,10 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
 	require.NoError(t, err)
 	defer dnsConn.Close()
 	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
-	_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
-
-	if err != nil && timeout == 0 {
-		t.Fatalf("Failed to get dns response %s\n", err.Error())
-	}
+	require.Eventually(t, func() bool {
+		_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
+		return err == nil || timeout != 0
+	}, 3*time.Second, 100*time.Millisecond, "Failed to get dns response")
```

Comment:
I don't think this is right, if the retry runs, then there will be a timeout recorded and we'll hit https://github.com/DataDog/datadog-agent/pull/20347/files#diff-e2871eefa3722b7a4b0bb1234dbbb5bb12305f81428382219b1b5d4ba5a67095R1062